File: db4on/tutorial/db4o-tutorial-chapters/src/com/db4o/f1/chapter1/ComplexQuery.cs
namespace com.db4o.f1.chapter1
{
    using com.db4o.query;

    public class ComplexQuery : Predicate
    {
    	public bool match(Pilot pilot)
    	{
	    	return pilot.Points>99
				&& pilot.Points<199
				|| pilot.Name=="Rubens Barrichello";
    	}
    }
}

File: db4on/tutorial/vb/db4o-tutorial-chapters/src/com/db4o/f1/chapter1/ComplexQuery.vb
Imports com.db4o.query
Namespace com.db4o.f1.chapter1
	Public Class ComplexQuery
	Inherits Predicate
		Public Function Match(ByVal pilot As Pilot) As Boolean
			Return pilot.Points > 99 AndAlso pilot.Points < 199 OrElse pilot.Name = "Rubens Barrichello"
		End Function

	End Class
End Namespace
